src/internal/connector/onedrive/restore.go

```diff
@@ -669,14 +714,70 @@ func createRestoreFolders(
 	return parentFolderID, nil
 }
 
+func copyBufferWithStallCheck(dst io.Writer, src io.Reader, stallTimeout time.Duration) (int64, error) {
+	buffer := make([]byte, copyBufferSize)
+
+	timer := time.NewTimer(stallTimeout)
+	defer timer.Stop()
+
+	type ce struct {
+		complete bool
+		err      error
+	}
+
+	totalCopied := int64(0)
+	cont := make(chan ce)
+
+	for {
+		go func() {
+			n, rerr := src.Read(buffer)
+			if rerr != nil && rerr != io.EOF {
+				cont <- ce{err: clues.Wrap(rerr, "reading data")}
+			}
+
+			_, werr := dst.Write(buffer[:n])
+			if werr != nil && werr != io.EOF {
+				cont <- ce{err: clues.Wrap(werr, "writing data")}
+			}
+
+			totalCopied += int64(n)
+
+			if rerr == io.EOF {
+				// Copy completed successfully
+				cont <- ce{complete: true}
+			}
+
+			cont <- ce{complete: false}
+		}()
+
+		timer.Reset(stallTimeout)
+
+		select {
+		case <-timer.C:
+			return 0, clues.New("copy stalled")
+		case in := <-cont:
+			if in.err != nil {
+				return 0, in.err
+			}
+
+			if in.complete {
+				return totalCopied, nil
+			}
+
+			// Continue copying
+			continue
```

Comment:
this can be removed since it's at the end of the case
```suggestion
```

Reply:
Hmm, that was added mostly to satisfy the linter. Having a block with just a comment was causing lint issues.